bitwarden/web
# [End User Vault Refresh] Vault - update cog menu

## Type of change


## Objective



* SG-31: Change the cog menu to 3-dots, make it always visible (not just on hover).
* While I'm here: change these menu items to use `<button>` elements instead of `<a>`. These are buttons that perform actions, not links that trigger navigation. This matters particularly for a11y.

## Code changes




Table row option menus and buttons are used throughout the app, and they're all the same. Therefore I've created the following components:
* **src/app/components/table-row-option-menu.component** - the bootstrap dropdown menu. Use it in a table row. Wrap it around the buttons you want in the menu.
* **src/app/components/table-row-option.component** - the individual buttons in the menu.

While these are small HTML snippets, they occur > 60 times throughout the app, and are a pain to update individually. They seemed like a good candidate for the component library.

All other changes implement these new components through every table in the app. 

This goes beyond just the End User Vault Refresh, however this change is also in the Org Admin Refresh work, so it makes sense to change all tables at once to keep a consistent look throughout the app.

## Screenshots


![Screen Shot 2022-03-10 at 2 25 14 pm](https://user-images.githubusercontent.com/31796059/157589279-a0dccb2b-fd50-4377-a7d8-30174c2e72b2.png)

## Testing requirements


Test all table option menus throughout the app - i.e.
* ensure no change in appearance
* ensure no change in functionality

## Before you submit
